svf/include/Util/SVFBugReport.h

```diff
@@ -0,0 +1,323 @@
+                             long long accessLowerBound, long long accessUpperBound):
+          GenericBug(bugType, eventStack), allocLowerBound(allocLowerBound),
+          allocUpperBound(allocUpperBound), accessLowerBound(accessLowerBound),
+          accessUpperBound(accessUpperBound){ }
+
+    cJSON *getBugDescription();
+    void printBugToTerminal();
+
+    /// ClassOf
+    static inline bool classof(const GenericBug *bug)
+    {
+        return bug->getBugType() == GenericBug::PARTIALBUFOVERFLOW || bug->getBugType() == GenericBug::FULLBUFOVERFLOW;
+    }
+};
+
+class FullBufferOverflowBug: public BufferOverflowBug{
+public:
+    inline FullBufferOverflowBug(EventStack eventStack,
+                             long long allocLowerBound, long long allocUpperBound,
+                             long long accessLowerBound, long long accessUpperBound):
+          BufferOverflowBug(GenericBug::FULLBUFOVERFLOW, eventStack, allocLowerBound,
+                            allocUpperBound, accessLowerBound, accessUpperBound){ }
+
+    /// ClassOf
+    static inline bool classof(const GenericBug *bug)
+    {
+        return bug->getBugType() == GenericBug::FULLBUFOVERFLOW;
+    }
+};
+
+class PartialBufferOverflowBug: public BufferOverflowBug{
+protected:
+    long long allocLowerBound, allocUpperBound, accessLowerBound, accessUpperBound;
+
+public:
+    inline PartialBufferOverflowBug( EventStack eventStack,
+                                 long long allocLowerBound, long long allocUpperBound,
+                                 long long accessLowerBound, long long accessUpperBound):
+          BufferOverflowBug(GenericBug::PARTIALBUFOVERFLOW, eventStack, allocLowerBound,
+                            allocUpperBound, accessLowerBound, accessUpperBound){ }
+
+    /// ClassOf
+    static inline bool classof(const GenericBug *bug)
+    {
+        return bug->getBugType() == GenericBug::PARTIALBUFOVERFLOW;
+    }
+};
+
+class NeverFreeBug : public GenericBug{
+public:
+    NeverFreeBug(EventStack bugEventStack):
+          GenericBug(GenericBug::NEVERFREE, bugEventStack){  };
+
+    cJSON *getBugDescription();
+    void printBugToTerminal();
+
+    /// ClassOf
+    static inline bool classof(const GenericBug *bug)
+    {
+        return bug->getBugType() == GenericBug::NEVERFREE;
+    }
+};
+
+class PartialLeakBug : public GenericBug{
+protected:
+    Set<std::string> conditionalFreePath;
+public:
+    PartialLeakBug(EventStack bugEventStack, Set<std::string> conditionalFreePath):
+          GenericBug(GenericBug::PARTIALLEAK, bugEventStack), conditionalFreePath(conditionalFreePath){ }
+
+    cJSON *getBugDescription();
+    void printBugToTerminal();
+
+    /// ClassOf
+    static inline bool classof(const GenericBug *bug)
+    {
+        return bug->getBugType() == GenericBug::PARTIALLEAK;
+    }
+};
+
+class DoubleFreeBug : public GenericBug{
+protected:
+    Set<std::string> doubleFreePath;
```

Comment:
Do we need this or put into the event stack?

Reply:
If you refer to doing so for DoubleFreeBug only, I think it's feasible, but may loss consistency with other bug classes like FileNeverCloseBug, NeverFreeBug, etc.

Reply:
And the reason for not doing so for other bug classes is that eventStack should be used to store events happening on the execution path from entry to the bug site, but not side effects like partial free paths and partial close paths.